apis/sls/v1alpha1/store_types.go

```diff
@@ -0,0 +1,98 @@
+/*
+
+ Copyright 2021 The Crossplane Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+*/
+
+package v1alpha1
+
+import (
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StoreSpec defines the desired state of SLS Store
+type StoreSpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+
+	// ForProvider field is where use set parameters for SLS Store
+	ForProvider StoreParameters `json:"forProvider"`
+}
+
+// StoreObservation is the representation of the current state that is observed.
+type StoreObservation struct {
+	// CreateTime is the time when the store was created
+	CreateTime uint32 `json:"createTime"`
+
+	// LastModifyTime is the time when the store was last modified
+	LastModifyTime uint32 `json:"lastModifyTime"`
+}
+
+// StoreStatus defines the observed state of SLS Store
+type StoreStatus struct {
+	xpv1.ResourceStatus `json:",inline"`
+	AtProvider          StoreObservation `json:"atProvider,omitempty"`
+}
+
+// StoreParameters define the desired state of an SLS store.
+type StoreParameters struct {
+	// SLS project name
+	// +kubebuilder:validation:MinLength:=3
+	// +kubebuilder:validation:MaxLength:=63
+	ProjectName string `json:"projectName"`
+
+	// The data retention period. Unit: days. If you set the value to 3650, the data is permanently stored
+	// +kubebuilder:validation:Minimum:=1
+	// +kubebuilder:validation:Maximum:=3650
+	TTL int `json:"ttl"`
+
+	// The number of shards
+	// +kubebuilder:validation:Minimum:=1
+	// +kubebuilder:validation:Maximum:=10
+	ShardCount int `json:"shardCount"`
+
+	// Specifies whether to enable automatic sharding. Default value: false.
+	// +optional
+	// +kubebuilder:default:=false
+	AutoSplit bool `json:"autoSplit"`
+
+	// The maximum number of shards for automatic sharding.
+	// +optional
+	MaxSplitShard int `json:"maxSplitShard"`
```

Comment:
By convention, we typically make optional parameters pointers and label with `omitempty`. This helps distinguish whether the field was set to the zero-value of the type (i.e. `false`) or whether it was omitted entirely.
```suggestion
	// Specifies whether to enable automatic sharding. Default value: false.
	// +optional
	// +kubebuilder:default:=false
	AutoSplit *bool `json:"autoSplit,omitempty"`

	// The maximum number of shards for automatic sharding.
	// +optional
	MaxSplitShard *int `json:"maxSplitShard,omitempty"`
```